backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -798,15 +807,43 @@ export class ExperimentService {
           partition.experiment = experimentDoc;
           return partition;
         });
-
+      
+      // creating queries docs
+      const promiseArray = [];
+      let queryDocsToSave =
+        (queries &&
+          queries.length > 0 &&
+          queries.map((query: Query) => {
+            promiseArray.push(this.metricRepository.findOne(query.metric));
+            // tslint:disable-next-line:no-shadowed-variable
+            const { createdAt, updatedAt, versionNumber, metric, ...rest } = query;
+            rest.experiment = experimentDoc;
+            rest.id = rest.id || uuid();
+            return rest;
+          })) ||
+        [];
+      if (promiseArray.length) {
+        const metricsDocs = await Promise.all([...promiseArray]);
+        queryDocsToSave = queryDocsToSave.map((queryDoc, index) => {
+          queryDoc.query.metric = metricsDocs[index];
+          return queryDoc;
+        });
+      }
+      
       // saving conditions and saving partitions
       let conditionDocs: ExperimentCondition[];
       let partitionDocs: ExperimentPartition[];
+      let queryDocs: Query[];
       try {
         [conditionDocs, partitionDocs] = await Promise.all([
           this.experimentConditionRepository.insertConditions(conditionDocsToSave, transactionalEntityManager),
           this.experimentPartitionRepository.insertPartitions(partitionDocsToSave, transactionalEntityManager),
         ]);
+        if (queryDocsToSave) {
+          queryDocs = await Promise.all([this.queryRepository.insertQueries(queryDocsToSave, transactionalEntityManager)]);
```

Comment:
Why have you used `Promise.all` if in the array only one element is present?

Reply:
Oh yes, nice catch. Will update this.